src/app.ts

```diff
@@ -13,14 +13,64 @@ interface TemperatureSummary {
   average: number
 }
 
+interface FullTemperatureSummary extends TemperatureSummary {
+  amountOfReadings: number;
+}
+
+interface ComputedTemperatureSummaries {
+  [city: string]: {
+    [time: string]: FullTemperatureSummary;
+  }
+}
+
+const summaries: ComputedTemperatureSummaries = {};
+
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  //add here your code
+  //compute temperature summaries, if there is new reading it updates or creates its summary
+
+  readings.forEach(reading => {
```

Comment:
why forEach instead of other loops?

Reply:
because each reading needs processing but no return value is needed